### PR TITLE
fix: add file protocol to solve windows pre rendering issues

### PR DESCRIPTION
**Pull request by angelfraga**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/analogjs/analog/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Which package are you modifying?

- [ ] vite-plugin-angular
- [x] vite-plugin-nitro
- [ ] astro-angular
- [ ] create-analog
- [ ] router
- [ ] platform
- [ ] content
- [ ] nx-plugin
- [ ] trpc

## What is the current behavior?

In windows pre rendering fails due next error

```
[nitro] [request error] [unhandled] Only URLs with a scheme in: file, data are supported by the default ESM loader. On Windows, absolute paths must be valid file:// URLs. Received protocol 'c:' 
```

Closes https://github.com/analogjs/analog/issues/688

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Comment by angelfraga** [comment]
@brandonroberts  while this change actually fixes the build in windows, I am aware this might not be the final fix.
I did not even tried out in WSL or Mac and probably we might need an additional OS check before prepending that.

File: packages/vite-plugin-nitro/src/lib/vite-plugin-nitro.ts [hunk 1]
```diff
@@ -104,9 +104,12 @@ export function nitro(options?: Options, nitroOptions?: NitroConfig): Plugin[] {
         }
 
         nitroConfig.alias = {
-          '#analog/ssr': normalizePath(
-            path.resolve(workspaceRoot, 'dist', rootDir, 'ssr/main.server')
-          ),
+          // This is not the final fix but start point to discuss a fix for windows
+          '#analog/ssr':
+            'file://' +
```
Comment (brandonroberts) [inline packages/vite-plugin-nitro/src/lib/vite-plugin-nitro.ts:109]: You can use this to check for windows

```ts
import os from 'node:os'

const isWindows = os.platform() === 'win32'
```

```suggestion
            (isWindows ? 'file://' : '') +
```

Comment (angelfraga) [reply packages/vite-plugin-nitro/src/lib/vite-plugin-nitro.ts:109]: Thanks, i will add it and check tests in Windows after work

**Comment by angelfraga** [comment]
@brandonroberts after running the test under windows with the fix for SSG actually does not fix all of them. In addition test snapshot have differencies between windows and other linux/mac.

But in the other hand this fix seems to solve the issue. So it would be great to have somebody else checking the fix under windows. 

In order to test the fix, just running the blog app in the repo is enough.

```
pnpm exec nx run blog-app:build --configuration=production
```

Before the fix

image

After the fix

image


Here is my env: 

Windows 11
Node 20.10.0

**Pull request merged**